addon/index.js

```diff
@@ -205,6 +205,7 @@ Router.prototype.on = function(name, f) {
 };
 
 Router.prototype.send = function(name, data, addon) {
+  this.mod.port.emit('from-addon-to-web', {type: name, data: data});
```

Comment:
During my debugging, exceptions thrown by the metrics ping broke the frontend client too. The exception was due to a typo, but having moved this doesn't hurt things either.